logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/ProtoContentMapper.kt

```diff
@@ -109,36 +110,46 @@ class ProtoContentMapperImpl : ProtoContentMapper {
-                        MessageContent.Unknown(encodedContent.data)
+                        MessageContent.Ignored
                     }
                 }
             }
-            is GenericMessage.Content.Ephemeral -> MessageContent.Unknown(encodedContent.data)
-            is GenericMessage.Content.External -> MessageContent.Unknown(encodedContent.data)
-            is GenericMessage.Content.Image ->
-                MessageContent.Unknown(encodedContent.data) // Deprecated in favor of GenericMessage.Content.Asset
+            is GenericMessage.Content.Ephemeral -> MessageContent.Ignored
+            is GenericMessage.Content.External -> MessageContent.Unknown(typeName, encodedContent.data)
+            is GenericMessage.Content.Image -> MessageContent.Ignored // Deprecated in favor of GenericMessage.Content.Asset
             is GenericMessage.Content.Hidden -> {
                 val hiddenMessage = genericMessage.hidden
                 if (hiddenMessage != null) {
                     MessageContent.DeleteForMe(hiddenMessage.messageId, hiddenMessage.conversationId, hiddenMessage.qualifiedConversationId)
                 } else {
                     kaliumLogger.w("Hidden message is null. Message UUID = $genericMessage.")
-                    MessageContent.Unknown(encodedContent.data)
+                    MessageContent.Ignored
                 }
             }
-            is GenericMessage.Content.Knock -> MessageContent.Unknown(encodedContent.data)
-            is GenericMessage.Content.LastRead -> MessageContent.Unknown(encodedContent.data)
-            is GenericMessage.Content.Location -> MessageContent.Unknown(encodedContent.data)
-            is GenericMessage.Content.Reaction -> MessageContent.Unknown(encodedContent.data)
+            is GenericMessage.Content.Knock -> MessageContent.Ignored
+            is GenericMessage.Content.LastRead -> MessageContent.Ignored
+            is GenericMessage.Content.Location -> MessageContent.Unknown(typeName, encodedContent.data)
+            is GenericMessage.Content.Reaction -> MessageContent.Ignored
             else -> {
                 kaliumLogger.w("Null content when parsing protobuf. Message UUID = $genericMessage.")
-                MessageContent.Unknown(encodedContent.data)
+                MessageContent.Unknown(typeName, encodedContent.data)
             }
         }
-        return ProtoContent(genericMessage.messageId, content)
+        val visibility = when (genericMessage.content) {
+            is GenericMessage.Content.Text,
+            is GenericMessage.Content.Asset,
+            is GenericMessage.Content.Calling,
+            is GenericMessage.Content.Composite,
+            is GenericMessage.Content.Edited,
+            is GenericMessage.Content.External,
+            is GenericMessage.Content.Location -> Message.Visibility.VISIBLE
+            is GenericMessage.Content.Deleted -> Message.Visibility.DELETED
+            else -> Message.Visibility.HIDDEN
```

Comment:
Why the default/else is hidden?

Reply:
We only want the specified above types to be visible, all other types shouldn't be displayed on conversation screen